src/vk/vk-contacts/vkcontactsyncadaptor.cpp

```diff
@@ -0,0 +1,1010 @@
+        urlQuery.addQueryItem("access_token", accessToken);
+        urlQuery.addQueryItem("v", QStringLiteral("5.21")); // version
+        requestUrl.setQuery(urlQuery);
+    } else {
+        requestUrl = QUrl(continuationRequest);
+    }
+
+    QNetworkRequest req(requestUrl);
+
+    // we're requesting data.  Increment the semaphore so that we know we're still busy.
+    incrementSemaphore(accountId);
+    QNetworkReply *reply = m_networkAccessManager->get(req);
+    if (reply) {
+        reply->setProperty("accountId", accountId);
+        reply->setProperty("accessToken", accessToken);
+        reply->setProperty("continuationRequest", continuationRequest);
+        reply->setProperty("lastSyncTimestamp", syncTimestamp);
+        reply->setProperty("startIndex", startIndex);
+        connect(reply, SIGNAL(finished()), this, SLOT(contactsFinishedHandler()));
+        connect(reply, SIGNAL(error(QNetworkReply::NetworkError)), this, SLOT(errorHandler(QNetworkReply::NetworkError)));
+        connect(reply, SIGNAL(sslErrors(QList<QSslError>)), this, SLOT(sslErrorsHandler(QList<QSslError>)));
+        m_apiRequestsRemaining[accountId] = m_apiRequestsRemaining[accountId] - 1;
+        setupReplyTimeout(accountId, reply);
+    } else {
+        SOCIALD_LOG_ERROR("unable to request data from VK account with id:" << accountId);
+        purgeSyncStateData(QString::number(accountId));
+        setStatus(SocialNetworkSyncAdaptor::Error);
+        decrementSemaphore(accountId);
+    }
+}
+
+void VKContactSyncAdaptor::contactsFinishedHandler()
+{
+    QNetworkReply *reply = qobject_cast<QNetworkReply*>(sender());
+    QByteArray data = reply->readAll();
+    int startIndex = reply->property("startIndex").toInt();
+    int accountId = reply->property("accountId").toInt();
+    QString accessToken = reply->property("accessToken").toString();
+    QDateTime lastSyncTimestamp = reply->property("lastSyncTimestamp").toDateTime();
+    bool isError = reply->property("isError").toBool();
+    reply->deleteLater();
+    removeReplyTimeout(accountId, reply);
+
+    SOCIALD_LOG_TRACE("received VK friends data for account:" << accountId << ":");
+    Q_FOREACH (const QString &line, QString::fromUtf8(data).split('\n', QString::SkipEmptyParts)) {
+        SOCIALD_LOG_TRACE(line);
+    }
+
+    if (isError) {
+        SOCIALD_LOG_ERROR("error occurred when performing contacts request for VK account:" << accountId);
+        purgeSyncStateData(QString::number(accountId));
+        setStatus(SocialNetworkSyncAdaptor::Error);
+        decrementSemaphore(accountId);
+        return;
+    } else if (data.isEmpty()) {
+        SOCIALD_LOG_ERROR("no contact data in reply from VK with account:" << accountId);
+        purgeSyncStateData(QString::number(accountId));
+        setStatus(SocialNetworkSyncAdaptor::Error);
+        decrementSemaphore(accountId);
+        return;
+    }
+
+    // parse the remote contact information from the response
+    QString continuationUrl;
```

Comment:
I have the same problem, need to test this with someone who has enough friends in VK.